simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
+                    than num_nodes//2
+
+                    :param relative_offset: relative to the width of the road, must be between -0.5 and 0.5
+                    :param absolute_offset: absolute, to account for line width
+                    :param output_number_of_points: number of outputs in list
+                    :return: list of width offsets
+                    """
+                    assert 0 < output_number_of_points < new_width_vals.__len__()//2, \
+                        "choose a valid number of output vals"
+                    assert -0.5 <= relative_offset <= 0.5, "relative offset must be between -0.5 and 0.5"
+                    assert -max(new_width_vals)/2 < absolute_offset < max(new_width_vals)/2, \
+                        "absolute offset must be smaller than half of the road"
+                    cutting_points = linspace(0, new_width_vals.__len__()-1, dtype=int, num=output_number_of_points)
+                    output_vals = list(map(lambda i: new_width_vals[i]*relative_offset + absolute_offset, cutting_points))
+                    return output_vals
+
+                # TODO Should this method coexist to _calculate_parallel_coords(offset: float, line_width: float)? Or
+                #  should the older method be integrated? (by simulating overloading using type()?)
+                #  The functionality is basically the same, except it works for changing widths and needs a list as
+                #  input. I tried to make the operation similar.
+                def _calculate_parallel_coords_list(offset: list, line_width: float) \
+                        -> Optional[List[Tuple[float, float, float, float]]]:
+                    """ calculates parallel coordinates of a road
+
+                    :param offset: list of offsets, must be smaller than num_nodes//2, should be equidistant
+                    :param line_width: specifies the width of the output coordinates
+                    :return: coordinates for the road
+                    """
+
+                    assert offset.__len__() < self.add_roads_to_scenario.num_nodes//2, \
+                        "there have to be less than half offset points of num_node for shapely LineStrings to work"
+
+                    original_line = LineString(zip(new_x_vals, new_y_vals))
+                    # num_points = original_line.length # seems to sometimes be trimmed to a bit more than 100?
+                    num_points = self.add_roads_to_scenario.num_nodes
+                    cutting_points = linspace(0, num_points-1, dtype=int, num=min(num_points, offset.__len__()))
+                    # extend the last point just a bit to get all nodes
+                    cutting_points[-1] = cutting_points[-1] + cutting_points [-1] - cutting_points[-2]
+
+                    offset_sub_lines = ()
+                    previous_p = 1
+                    i = 0
+                    # split road into multiple pieces
+                    for p in cutting_points:
```

Comment:
The method is very long and has a very high level of indentations. Please try to split it such that its structure improves and makes it easier to understand.

Reply:
I tried to split the method, this resulted in long parameter lists